### PR TITLE
TELCODOCS-655: D/S Docs & RN: AI SaaS - Installing with the AI - Updates to existing content

### PR DESCRIPTION
**Pull request by johnwilkins**
Added a module for how to use the Assisted Installer.

Fixes: [TELCODOCS-655](https://issues.redhat.com//browse/TELCODOCS-655)

See https://issues.redhat.com/browse/TELCODOCS-655 for additional details.

Preview URL: http://157.131.167.205/TELCODOCS-655/installing/installing_on_prem_assisted/installing-on-prem-assisted.html

For release(s): 4.10, 4.11
Signed-off-by: John Wilkins <jowilkin@redhat.com>

### CODE REVIEW

**Comment by johnwilkins** [comment]
@jelkosz @racedo Can you review the PR for the contents of https://issues.redhat.com//browse/TELCODOCS-655? There are several other JIRA tickets too. This is just the first one.

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// This is included in the following assemblies:
+//
+// installing-on-prem-assisted.adoc
+:_content-type: CONCEPT
+
+[id='how-to-use-the-assisted-installer_{context}']
+= How to use the Assisted Installer
+
+{product-title} {ai-full} is a user-friendly installation solution offered on link:http://console.redhat.com[console.redhat.com]. {ai-full} supports the various deployment platforms with a focus on bare metal. It comprises two key components:
```
Comment (racedo) [inline modules/assisted-installer-how-to-use-the-assisted-installer.adoc:9]: I'd link here the actual AI link you added at the bottom to access AI directly.

Also, when referring to cloud.redhat.com I'd call it as Red Hat Hybrid Cloud Console

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  - Ignition files
+  - The installation configuration
+  - A discovery ISO
+  - The web user interface
+  - The REST API
+
+* *{ai-full}*: The {ai-full} provides the installation functionality, and it has following advantages:
+  - *Web user interface:* The web user interface performs cluster installation without the user having to create the installation configuration file manually.
+  - *No bootstrap node:* A bootstrap node is not required when installing with the {ai-full}. The bootstrapping process executes on a random node within the cluster.
+  - *Simplified deployment:* Deployment does not require in-depth knowledge of {product-title}.
```
Comment (jelkosz) [inline modules/assisted-installer-how-to-use-the-assisted-installer.adoc:21]: I would add also "rich pre flight validation and reasonable defaults"

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  - A flexible API
+  - You can deploy Single Node OpenShift (SNO)
+  - You can install {VirtProductName} and {rh-storage} (formerly OpenShift Container Storage) from the web user interface.
+
+The {ai-full} is a great solution for:
+
+* Users installing {product-title} on premises in a connected environment.
+* Users without the need for strictly reproducible results. Developers frequently update the installable versions to the latest tested z-stream releases. Some older versions remain available, if needed.
+* Users that don't need automation. {ai-full} is great for first clusters, demos, and proof of concept projects.
```
[the remark this answers is not in the document]
Comment (romfreiman) [reply modules/assisted-installer-how-to-use-the-assisted-installer.adoc:30]: I agree. This is not the way we want to position AI. Good catch. Users can (and do) build automation around ai apis. 

Same being said about 'no need for strictly repoducible results' - where does it comes from?

**Comment by romfreiman** [comment]
/hold
Untill I review on Sunday

**Comment by romfreiman** [comment]
What is the separation between Assisted Service and Assisted Installer? Where does it comes from? Even I cannot explain the differences. Why is it required and how does it helps / affects the users?

**Comment by romfreiman** [comment]
The Assisted Installer is a great solution for:
* Users installing OpenShift Container Platform on premises in a connected environment.
* Users that are interested to have pre flights validations in order to have a higher certainty of a successful installation.
* Users that are interested to use the latest and greatest OpenShift, without requirement to run the installer locally in their data center. The service frequently updates the installable versions to the latest tested z-stream releases. Some older versions remain available, if needed.
* Users that are interested to build an automation for deploying OpenShift, without requirement to run the installer locally in their data center

feel free to iterate over the above text.

Please remove the *strictly reproducible results*, *Users that don’t need automation. Assisted Installer is great for first clusters, demos, and proof of concept projects.* - not sure where does it comes from

**Comment by romfreiman** [comment]
API support for Assisted Installer - why this is the 2nd bullet in the page, and not the last one?

**Comment by romfreiman** [comment]
Lets mention vsphere platform integration here as well

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (avishayt) [reply modules/assisted-installer-how-to-use-the-assisted-installer.adoc:30]: Instead of "Users that don't need automation", maybe you meant "users that don't have automation".  So it's great for first clusters, demos, PoCs as you said, but you can definitely add automation as the comments above have said.

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc (hunk 2 above)
Comment (gamli75) [reply modules/assisted-installer-how-to-use-the-assisted-installer.adoc:21]: what about mentioning a few of the main features:
SNO
vsphere
dual stack
static ip

**Comment by johnwilkins** [comment]
> What is the separation between Assisted Service and Assisted Installer? Where does it comes from? Even I cannot explain the differences. Why is it required and how does it helps / affects the users?

@romfreiman It came from this doc, which I was asked to port to the main docs: https://cloud.redhat.com/blog/how-to-use-the-openshift-assisted-installer  It may have a bunch of engineering detail that isn't needed for end users. The other source of information is this slide: https://docs.google.com/presentation/d/1slEK4rynwQwdMNll_l2_Qb7gWSHq2NJceVkKwzBt3Xc/edit#slide=id.g115a827501f_0_20

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc (hunk 1 above)
Comment (johnwilkins) [reply modules/assisted-installer-how-to-use-the-assisted-installer.adoc]: I've added both to the sentence.

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc (hunk 2 above)
Comment (johnwilkins) [reply modules/assisted-installer-how-to-use-the-assisted-installer.adoc]: I've added those comments. Let me know what you think.

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (johnwilkins) [reply modules/assisted-installer-how-to-use-the-assisted-installer.adoc]: @jelkosz I've updated the doc to make it clearer that you can use the API to automate later so it doesn't sound like AI is just for trivial clusters.

**Comment by romfreiman** [comment]
@johnwilkins ok, dont list to the one who wrote that slidedeck :)

**Comment by romfreiman** [comment]
_executes on a random node within the cluster._ - I would remove the word _random_

**Comment by romfreiman** [comment]
_Installing OpenShift Container Platform on premises in a connected environment._ - can we add something about proxy here as well - i want to emphasis that behind proxy is counted as connected as well

**Comment by romfreiman** [comment]
Looks good IMO. Thanks for fast iteration

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc [hunk 4]
```diff
@@ -0,0 +1,44 @@
+// This is included in the following assemblies:
+//
+// installing-on-prem-assisted.adoc
+:_content-type: CONCEPT
+
+[id="how-to-use-the-assisted-installer_{context}"]
+= How to use the Assisted Installer
+
+The {product-title} link:https://console.redhat.com/openshift/assisted-installer/clusters/~new[{ai-full}] is a user-friendly installation solution offered on the link:http://console.redhat.com[Red Hat Hybrid Cloud Console]. The {ai-full} supports the various deployment platforms with a focus on bare metal.
```
Comment (avishayt) [inline modules/assisted-installer-how-to-use-the-assisted-installer.adoc:9]: @romfreiman WDYT?

```suggestion
The {product-title} link:https://console.redhat.com/openshift/assisted-installer/clusters/~new[{ai-full}] is a user-friendly installation solution offered on the link:http://console.redhat.com[Red Hat Hybrid Cloud Console]. The {ai-full} supports the various deployment platforms with a focus on bare metal and vSphere.
```

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+The {ai-full} provides installation functionality as a service. This software-as-a-service (SaaS)  approach has the following advantages:
```
Comment (avishayt) [inline modules/assisted-installer-how-to-use-the-assisted-installer.adoc:11]:
```suggestion
The {ai-full} provides installation functionality as a service. This software-as-a-service (SaaS) approach has the following advantages:
```

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+* *Web user interface:* The web user interface performs cluster installation without the user having to create the installation configuration files manually.
+* *No bootstrap node:* A bootstrap node is not required when installing with the {ai-full}. The bootstrapping process executes on a node within the cluster.
+* *Hosting:* The {ai-full} hosts:
+  - Ignition files
+  - The installation configuration
+  - A discovery ISO
+  - The installer
+* *Simplified deployment:* Deployment does not require in-depth knowledge of {product-title}. The {ai-full} provides reasonable defaults. Additionally, the {ai-full} performs rich pre-flight validation before installation to catch configuration-related installation errors before they occur.
```
Comment (avishayt) [inline modules/assisted-installer-how-to-use-the-assisted-installer.adoc:20]:
```suggestion
* *Simplified deployment:* Deployment does not require in-depth knowledge of {product-title}. The {ai-full} provides reasonable defaults. Additionally, the {ai-full} performs rich pre-flight validations before installation to catch configuration-related installation errors before they occur.
```

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+* *Advanced networking:* The {ai-full} supports IPv4/IPv6 dual stack networking, NMState-based static IP addressing, and an HTTP/S proxy.
+* *REST API:* The {ai-full} has a REST API, enabling automation.
```
Comment (avishayt) [inline modules/assisted-installer-how-to-use-the-assisted-installer.adoc:22]: Do you want to add some links?
- This is the swagger that defines the API: https://github.com/openshift/assisted-service/blob/master/swagger.yaml
- Doc on how to use it: https://github.com/openshift/assisted-service/blob/master/docs/cloud.md

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+* *Multiple installation scenarios:* The {ai-full} supports numerous scenarios, including:
+  - Installing {product-title} on premises in a connected environment, including with an optional HTTP/S proxy.
+  - Installing Single Node OpenShift (SNO)
+  - Installing on vSphere
+  - Installing {VirtProductName} and {rh-storage} (formerly OpenShift Container Storage) from the web user interface
+  - Installing first clusters, demos, and proof of concept projects that don't have automation. Users can add automation to the {ai-full} through the API later
+* *Streamlined installation workflow:* The {ai-full} provides the installer as a service, which:
+  - Eliminates the need to install and run the installer locally
+  - Ensures the latest version of the installer up to the latest tested z-stream releases. Older versions remain available, if needed
+  - Enables building automation through the API without the need to run the installer locally
```
Comment (avishayt) [inline modules/assisted-installer-how-to-use-the-assisted-installer.adoc:42]: These 3 high-level bullet points don't really follow from the title.  For example, "AI is a great solution for pre-installation validations"?  I recommend moving the first and third bullets (validations and streamlined installation workflow) up to where you discuss "Simplified deployment".  I recommend moving the second bullet up to the beginning of the file where you write "The {ai-full} supports the various deployment platforms with a focus on bare metal."

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+  - Ensuring connectivity to the registry
+  - Ensuring time synchronization between cluster nodes
+  - Verifying that the cluster nodes meet the minimum hardware requirements
+  - Validating the installation configuration parameters
+* *Multiple installation scenarios:* The {ai-full} supports numerous scenarios, including:
+  - Installing {product-title} on premises in a connected environment, including with an optional HTTP/S proxy.
+  - Installing Single Node OpenShift (SNO)
+  - Installing on vSphere
+  - Installing {VirtProductName} and {rh-storage} (formerly OpenShift Container Storage) from the web user interface
+  - Installing first clusters, demos, and proof of concept projects that don't have automation. Users can add automation to the {ai-full} through the API later
```
Comment (avishayt) [inline modules/assisted-installer-how-to-use-the-assisted-installer.adoc:38]:
```suggestion
The {ai-full} supports installing {product-title} on premises in a connected environment, including with an optional HTTP/S proxy.  It can install:
  - Highly available {product-title} or Single Node OpenShift (SNO)
  - Bare-metal or vSphere
  - Optionally installing {VirtProductName} and {rh-storage} (formerly OpenShift Container Storage) from the web user interface
  
The user interface provides an intuitive interactive flow where automation does not exist or is not required (e.g., first clusters, demos, and proof of concept projects). Users may also automate installations using the REST API.
```

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc [hunk 10]
```diff
@@ -0,0 +1,44 @@
+// This is included in the following assemblies:
+//
+// installing-on-prem-assisted.adoc
+:_content-type: CONCEPT
+
+[id="how-to-use-the-assisted-installer_{context}"]
+= How to use the Assisted Installer
```
Comment (avishayt) [inline modules/assisted-installer-how-to-use-the-assisted-installer.adoc:7]: This doc isn't really about "how" to use the Assisted Installer, more like when or why.  Maybe just "Using the Assisted Installer"?

**Comment by avishayt** [comment]
@johnwilkins I had a lot of comments, so I made a doc with a suggested change so it's easier to follow: https://docs.google.com/document/d/1k2ju_KZg0Gaa78Opf4Ph3hy_kK5Gi0NzvuMC8vUkPQQ/edit?usp=sharing

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc (hunk 7 above)
Comment (johnwilkins) [reply modules/assisted-installer-how-to-use-the-assisted-installer.adoc:22]: They ask us not to link to github docs. So we'd probably just bring these downstream at some point.

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc (hunk 9 above)
Comment (johnwilkins) [reply modules/assisted-installer-how-to-use-the-assisted-installer.adoc]: I made the changes. However, I omitted the exempli gratia, because it sort of re-introduces the idea that AI isn't suitable for production clusters. I think it reads fine without those examples. What do you folks think?

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc (hunk 4 above)
Comment (johnwilkins) [reply modules/assisted-installer-how-to-use-the-assisted-installer.adoc]: Done.

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc (hunk 5 above)
Comment (johnwilkins) [reply modules/assisted-installer-how-to-use-the-assisted-installer.adoc]: Fixed.

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc (hunk 6 above)
Comment (johnwilkins) [reply modules/assisted-installer-how-to-use-the-assisted-installer.adoc]: Fixed, but I consolidated this with streamlined workflow bullet.

File: modules/assisted-installer-how-to-use-the-assisted-installer.adoc (hunk 8 above)
Comment (johnwilkins) [reply modules/assisted-installer-how-to-use-the-assisted-installer.adoc]: Done

**Comment by avishayt** [comment]
Changes look good, thanks!

**Comment by romfreiman** [comment]
_while still running pre-flight _ - I dont think _still_ is relevant here. no other installer does it

**Comment by johnwilkins** [comment]
> _while still running pre-flight _ - I dont think _still_ is relevant here. no other installer does it

@romfreiman That phrase isn't in the latest cut. Can you have a look at the preview again? Also, if it's satisfactory, can your remove the do-not-merge flag?

**Review by avishayt — APPROVED** [review]

File: modules/assisted-installer-using-the-assisted-installer.adoc [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+* *No bootstrap node:* A bootstrap node is not required when installing with the {ai-full}. The bootstrapping process executes on a node within the cluster.
+* *Hosting:* The {ai-full} hosts:
+  - Ignition files
+  - The installation configuration
+  - A discovery ISO
+  - The installer
+* *Streamlined installation workflow:* Deployment does not require in-depth knowledge of {product-title}. The {ai-full} provides reasonable defaults and provides the installer as a service, which:
+  - Eliminates the need to install and run the {product-title} installer locally
+  - Ensures the latest version of the installer up to the latest tested z-stream releases. Older versions remain available, if needed
+  - Enables building automation through the API without the need to run the {product-title} installer locally
```
Comment (jldohmann) [inline modules/assisted-installer-using-the-assisted-installer.adoc:23]: ISG says to avoid using "through" as a preposition:
```suggestion
  - Enables building automation by using the API without the need to run the {product-title} installer locally.
```
The second part of this sentence "without the need ..." is already highlighted in another sub-item above this one. Does it need to be restated here, or can it be removed? E.g.
```suggestion
  - Enables building automation by using the API.
```

File: modules/assisted-installer-using-the-assisted-installer.adoc [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+  - Ensuring connectivity to the registry
+  - Ensuring time synchronization between cluster nodes
+  - Verifying that the cluster nodes meet the minimum hardware requirements
+  - Validating the installation configuration parameters
+* *REST API:* The {ai-full} has a REST API, enabling automation.
+
+The {ai-full} supports installing {product-title} on premises in a connected environment, including with an optional HTTP/S proxy.  It can install:
+
+* Highly available {product-title} or Single Node OpenShift (SNO)
+* On bare metal or vSphere with full platform integration, or other virtualization platforms without integration
```
Comment (jldohmann) [inline modules/assisted-installer-using-the-assisted-installer.adoc:37]: This was a little bit hard to parse, because it describes a place to install something vs a thing to be installed like the other items. For clarity, it might make sense to have it begin without the preposition, so that it matches more of what the other items are trying to say:
```suggestion
* {product-title} on bare metal or vSphere with full platform integration, or other virtualization platforms without integration
```

**Review by jldohmann — COMMENTED** [review]
Some minor ISG nits, but otherwise LGTM. I also left some suggestions.

**Review by jldohmann — APPROVED** [review]
